### PR TITLE
Make RNNs generate dropout masks whenever they are called

### PR DESCRIPTION
**Pull request by taehoonlee**
This PR addresses errors #8976 and follows up #9006.

### CODE REVIEW

**Comment by fchollet** [comment]
This would break recurrent dropout. The point of storing dropout masks as layer attributes is to be able to reuse them across different timesteps (i.e. across different calls to the cell).

We should reset them with every call of the batch-level RNN layer, not every call of the cell layer.

**Comment by taehoonlee** [comment]
@fchollet, Would you please check this version? Now we can run the following codes.

```python
import numpy as np
import tensorflow as tf
from keras.models import Sequential, Model
from keras.layers import Input, Dense, LSTM, Conv2D
from keras.utils import multi_gpu_model

lstm = LSTM(3, dropout=0.2, recurrent_dropout=0.3)

a = Input(batch_shape=(3, 5, 10))
b = Input(batch_shape=(4, 6, 10))
c = lstm(a)
d = lstm(b)

x_train1 = np.random.random((300, 5, 10))
y_train1 = np.random.random((300, 3))
x_train2 = np.random.random((400, 6, 10))
y_train2 = np.random.random((400, 3))

model1 = Model(a, c)
model1.compile(loss='mse', optimizer='adam')
model1.fit(x_train1, y_train1, epochs=1, batch_size=3)
model2 = Model(b, d)
model2.compile(loss='mse', optimizer='adam')
model2.fit(x_train2, y_train2, epochs=1, batch_size=4)
```
